src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/CollectQualityYieldMetricsSpark.java

```diff
@@ -0,0 +1,198 @@
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.Serializable;
+
+/**
+ * Command line program to calibrate quality yield metrics
+ *
+ * This is the Spark version.
+ */
+@CommandLineProgramProperties(
+        summary = "Collects quality yield metrics, a set of metrics that quantify the quality and yield of sequence data from a " +
+                "SAM/BAM input file.",
+        oneLineSummary = "Collects a set of metrics that quantify the quality and yield of sequence data from the provided SAM/BAM",
+        programGroup = SparkProgramGroup.class
+)
+public final class CollectQualityYieldMetricsSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    @Argument(shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            doc = "The metrics file to write with quality yield metrics.")
+    public String out;
+
+    @Argument(shortName = StandardArgumentDefinitions.USE_ORIGINAL_QUALITIES_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.USE_ORIGINAL_QUALITIES_LONG_NAME,
+            doc = "If available in the OQ tag, use the original quality scores " +
+                    "as inputs instead of the quality scores in the QUAL field.")
+    public boolean useOriginalQualities = false;
+
+
+    /** A set of metrics used to describe the general quality of a BAM file */
+    public static final class QualityYieldMetrics extends MetricBase implements Serializable{
+        private static final long serialVersionUID = 1;
+
+        private boolean useOriginalQualities;
+        private boolean done;
+
+        //NOTE: this class must have a 0-arg constructor so we need to use a setter for this field.
+        //Also, we want to be able to easily construct usable instances so the setter returns 'this' so we can say
+        //new QualityYieldMetrics().setUseOriginalQualities(true) in the aggregate call
+        public QualityYieldMetrics setUseOriginalQualities(final boolean useOriginalQualities){
+            this.useOriginalQualities = useOriginalQualities;
+            return this;
+        }
+
+        /** The total number of reads in the input file */
+        public int TOTAL_READS;
```

Comment:
I know. The problem is that those metrics are printed out to this file that is consumed by some Broad tools downstream and those expect this format. I'll add comments about this throughout